### ISSUE TITLE
ENT-041: Core metrics collector with trueno SIMD aggregation

### ISSUE THREAD

**Issue by noahgift** [paiml/entrenar#14]
## Phase 1 of #13

Implement `MetricsCollector` with low-overhead SIMD metric aggregation.

## Deliverables

- [ ] `src/monitor/mod.rs` - Public API
- [ ] `src/monitor/collector.rs` - MetricsCollector implementation
- [ ] Basic metrics: loss, accuracy, learning_rate, gradient_norm
- [ ] JSON export for immediate visibility
- [ ] Running statistics via trueno::Vector (mean, std, min, max)

## API

```rust
use entrenar::monitor::{MetricsCollector, Metric};

let mut collector = MetricsCollector::new();
collector.record(Metric::Loss, 0.5);
collector.record(Metric::Accuracy, 0.85);
let stats = collector.summary();
```

## Acceptance Criteria

- [ ] Overhead < 1% of training time
- [ ] Test coverage > 90%
- [ ] Supports batch recording for efficiency

**Estimate:** 8 hours
**Spec:** docs/specifications/training-monitoring-spec-realtime-and-diagnostics.md
